cdc/sink/dmlsink/mq/dispatcher/topic/expression.go

```diff
@@ -21,9 +21,12 @@ import (
 )
 
 var (
+	// hardCodeTopicNameRe is used to match a topic name which is hard code in the config
+	hardCodeTopicNameRe = regexp.MustCompile(`^([A-Za-z0-9\._\-]+)$`)
+
 	// topicNameRE is used to match a valid topic expression
 	topicNameRE = regexp.MustCompile(
-		`^[A-Za-z0-9\._\-]*\{schema\}([A-Za-z0-9\._\-]*\{table\})?[A-Za-z0-9\._\-]*$`,
+		`^[A-Za-z0-9\._\-]*(\{schema\})?([A-Za-z0-9\._\-]*\{table\})?[A-Za-z0-9\._\-]*$`,
```

Comment:
whether the empty case is covered by this expression?

Reply:
what do you mean by the `empty` case ?

Reply:
Yes it also covers the empty case. But if user enter "" as the topic, the default topic is used directly, won't hit this logic.